src/store/models/designer.ts

```diff
@@ -213,6 +231,101 @@ const designerModel: DesignerModel = {
       }
     });
   }),
+  renameLightningNode: thunk(
+    async (actions, { nodeId, name, backendName }, { getState, getStoreActions }) => {
+      const { allCharts, activeId } = getState();
+      const chart = allCharts[activeId];
+
+      // Ensure the node exists
+      const node = chart.nodes[nodeId];
+      if (!node) {
+        throw new Error(`Lightning node with id ${nodeId} not found.`);
+      }
+
+      // Update the node chart
+      const updatedChart = {
+        ...chart,
+        nodes: {
+          ...chart.nodes,
+          [name]: {
+            ...chart.nodes[nodeId],
+            id: name,
+          },
+        },
+      };
+
+      // Update the state with the modified chart
+      allCharts[activeId] = updatedChart;
+      // remove the old node with previous key
+      getStoreActions().designer.removeNode(nodeId);
+      getStoreActions().designer.updateBackendLink({
+        lnName: name,
+        backendName: backendName,
+      });
```

Comment:
Instead of calling `updateBackendLink`, this function should loop over all of `chart.links` and check if the `link.from.nodeId` or `link.to.nodeId` is equals the old name. If it does, then change the `nodeId` to the new name. This should fix the missing links.